python/ambassador/ir/ir.py

```diff
@@ -729,13 +729,17 @@ def save_secret_info(self, aconf):
             #
             # (We include 'user_key' here because ACME private keys use that, and they
             # should not generate errors.)
-            if aconf_secret.get('tls_crt') or aconf_secret.get('cert-chain_pem') or aconf_secret.get('user_key'):
+            # (We include 'crl_pem' here because CRL secrets use that, and they
+            # should not generate errors.)
+            if aconf_secret.get('tls_crt') or aconf_secret.get('cert-chain_pem') or aconf_secret.get('user_key') or aconf_secret.get('crl_pem'):
                 secret_info = SecretInfo.from_aconf_secret(aconf_secret)
                 secret_name = secret_info.name
                 secret_namespace = secret_info.namespace
 
                 self.logger.debug('saving "%s.%s" (from %s) in secret_info', secret_name, secret_namespace, secret_key)
                 self.secret_info[f'{secret_name}.{secret_namespace}'] = secret_info
+            else:
+                self.logger.debug('not saving secret_info from %s because there is no public half', secret_key)
```

Comment:
1. should this be Info or Warning? I see the  previous line above is debug but I would think the fact that this doesn't exist would want to be seen by a developer rather than hidden until they turned on debug logging?

2. Is `secret_key` something we should be print to logs? Again, I see it was on the previous log line but just want to make sure there is no security concern there?

Reply:
Good questions!
1. I went with `debug` since this line did not exist before and missing secrets are reported later in code execution. It was useful while debugging KAT tests (which run in debug).
2. I believe the `secret_key` in this context is the unique Key in the secrets dict. It references the secret name, not the potential sensitive secret private key.